src/libraries/System.IO.Hashing/src/System/IO/Hashing/Crc32.Arm.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using ArmCrc = System.Runtime.Intrinsics.Arm.Crc32;
+
+namespace System.IO.Hashing
+{
+    public partial class Crc32
+    {
+        private static uint UpdateScalarArm64(uint crc, ReadOnlySpan<byte> source)
+        {
+            Debug.Assert(ArmCrc.Arm64.IsSupported, "ARM CRC support is required.");
+
+            // Compute in 8 byte chunks
+            if (source.Length >= sizeof(ulong))
+            {
+                ReadOnlySpan<ulong> longSource = MemoryMarshal.Cast<byte, ulong>(source);
```

Comment:
This should really use `ReadUnaligned` since there is no guarantee that things are "properly aligned" otherwise.

Reply:
Rewritten to use ref byte and Unsafe.ReadUnaligned